pkg/runtime/runtime.go

```diff
@@ -2649,32 +2648,42 @@ func (a *DaprRuntime) cleanSocket() {
 func (a *DaprRuntime) Shutdown(duration time.Duration) {
 	// Ensure the Unix socket file is removed if a panic occurs.
 	defer a.cleanSocket()
-
-	log.Info("dapr shutting down.")
-
+	log.Info("Dapr shutting down")
 	log.Info("Stopping PubSub subscribers and input bindings")
 	a.stopSubscriptions()
 	a.stopReadingFromBindings()
-	a.cancel()
+
+	log.Info("Initiating actor shutdown")
 	a.stopActor()
-	log.Info("Stopping Dapr APIs")
-	for _, closer := range a.apiClosers {
-		if err := closer.Close(); err != nil {
-			log.Warnf("error closing API: %v", err)
-		}
-	}
 
-	shutdownCtx, shutdownCancel := context.WithCancel(context.Background())
+	// Flush and shutdown the tracing provider.
+	shutdownCtx, shutdownCancel := context.WithCancel(a.ctx)
+	defer shutdownCancel()
 	go func() {
 		if a.tracerProvider != nil {
```

Comment:
I don't think we can stop the trace provider *before* the APIs, otherwise we'll be losing traces from the API calls that come in after.

Reply:
Still think the trace provider should be stopped synchronously (not in a goroutine), with a short timeout, afterthe APIs

Reply:
https://github.com/dapr/dapr/blob/1f75b16a21344188d41dd9d7ec39b87a8df50cd4/pkg/runtime/runtime.go#L2648-L2671

Reply:
@ItalyPaleAle - this comment is resolved. Can you please check and mark this as resolved? Thanks.